torch/csrc/jit/tracer.h

```diff
@@ -131,22 +131,22 @@ TORCH_API void addInputs(
     const std::unordered_map<K, V>& value);
 
 template<typename T>
-void addInputs(
+TORCH_API void addInputs(
```

Comment:
What about moving these definitions into the source file `tracer.cpp`?

Reply:
Templates have to be in the header. What do you mean?

Reply:
Oh sorry, I messed it up. But the compiler error is real. I believe these warnings will only occur when TORCH_API is `__declspec(dllimport)`. So to solve this error, we will need to split `TORCH_API` into `TORCH_EXPORT` and `TORCH_IMPORT` and mark this one with `TORCH_EXPORT`. (`TORCH_EXPORT` should be empty when we are actually importing symbols.)

Reply:
I assumed TORCH_API was already defined like this. Only those two were giving warnings.

The compiler (msvc) already defines <target>_EXPORTS.

https://github.com/pytorch/pytorch/blob/master/torch/csrc/WindowsTorchApiMacro.h